File: ParseScript1.scala
object ParseSpinn3r{
	import java.io.{BufferedReader,FileReader,BufferedWriter,FileWriter}
	
	class Tweet{
		var id:String=""
		var author:String=""
		var text:String=""
		var date:String=""
		var lang:String=""
		var ready:Boolean=false
	}
	
	def main(args: Array[String]){
		val fileReader = new BufferedReader(new FileReader(args(0)))
		val fileWriter = new BufferedWriter(new FileWriter("output.xml"))
		var CurrentTweet = new Tweet
		var go=true
		var line=""
		fileWriter.write("<add>\n");
		while(go){
			line = fileReader.readLine
			if(line!=null){
				if(line.compareTo("----")==0){
					if(CurrentTweet.ready){
						fileWriter.write("<doc>\n")
						fileWriter.write("\t<field name=\"id\">"+CurrentTweet.id+"</field>\n")
						fileWriter.write("\t<field name=\"author\">"+CurrentTweet.author+"</field>\n")
						fileWriter.write("\t<field name=\"date\">"+CurrentTweet.date+"</field>\n")
						fileWriter.write("\t<field name=\"lang\">"+CurrentTweet.lang+"</field>\n")
						fileWriter.write("\t<field name=\"text\">"+CurrentTweet.text+"</field>\n")
						fileWriter.write("</doc>\n")
						fileWriter.flush
					}
					CurrentTweet=new Tweet
				}
				else{
					try{
						(line.split(":")(0)) match{
							case ("link") =>
								CurrentTweet.id=line.substring(line.indexOf("statuses/")+9)
							case ("title") =>
								CurrentTweet.text=line.substring(line.indexOf(":")+1).trim.replaceAll("&","and").replaceAll("<","(lt)").replaceAll(">","(gt)");
							case ("published") =>
								CurrentTweet.date=line.substring(line.indexOf(":")+1).trim
							case ("author name") =>
								CurrentTweet.author=line.substring(line.indexOf(":")+1).trim
							case ("lang") =>
								CurrentTweet.lang=line.substring(line.indexOf(":")+1).trim
								CurrentTweet.ready=true
							case trash:String =>
						}
					}
					catch{
						case e:Exception=>
							// println(line)
					}
				}
			}
			else{
				go=false
			}
		}
		fileWriter.write("</add>\n")
		fileWriter.flush
		fileWriter.close
	}
}

File: ParseScript1.pl
#!/usr/local/bin/perl
open (FILE, 'rawdata.txt');
my $records = 0;
my $flushLevel = 100000;
my $fileNum = 0;
my $fileStr=sprintf("%04d", $fileNum);
my $ready = 0;
my $lines = 0;

my $id="";
my $text="";
my $date="";
my $name="";
my $lang="";

open my $outf, '>', "myOutput$fileStr.xml";

while (<FILE>) {
	if ($lines==0){
		# print "<add>\n";
		print $outf "<add>\n";
#		print $fileNum;
#		print ": ";
		$lines++;
	}
	
	chomp;
	my $line = $_;
	if($line =~ m/^link:/){
		if($line =~ m/^.*statuses\/[\d]*$/){
			$line =~ s/^.*statuses\/([\d]*)$/$1/;		
		}
		else{
			$line = 0;		
		}
		# $line =~ s/[^\d]//g;
		$id = $line;
	}
	elsif($line =~ m/^title:/){
		$line =~ s/^.*:[ ]*(.*)$/$1/;
		$line =~ s/\&/$1(and)$2/g;
		$line =~ s/\</$1(lt)$2/g;
		$line =~ s/\>/$1(gt)$2/g;
		$text = $line;
	}
	elsif($line =~ m/^pubDate:/){
		$line =~ s/^.*:[ ]+(.*)$/$1/;
		$date = $line;
	}
	elsif($line =~ m/^author name:/){
		$line =~ s/^.*:[ ]*(.*)$/$1/;
		$line =~ s/\&/$1(and)$2/g;
		$line =~ s/\</$1(lt)$2/g;
		$line =~ s/\>/$1(gt)$2/g;
		$name = $line;
	}
	elsif($line =~ m/^lang:/){
		$line =~ s/^.*:[ ]*(.*)$/$1/;
		$lang = $line;
		$ready = 1;
	}
	
	if($ready==1){
		$records++;
		$out=<<END;
<doc>
	<field name="id">$id</field>
	<field name="text">$text</field>
	<field name="date">$date</field>
	<field name="name">$name</field>
	<field name="lang">$lang</field>
</doc>
END
#		print "$records\n";
		# print ".";
		print $outf $out;
		$ready=0;
	}
	if ($records > $flushLevel){
		$fileNum+=1;
		$fileStr=sprintf("%04d", $fileNum);
		print $outf "</add>\n";
		$|++;
		open my $outnew, '>', "myOutput$fileStr.xml";
		close($outf);
		$outf=$outnew;
		$lines=0;
		$records=0;
	}
}
close (FILE);
print $outf "</add>\n";
close($outf);
